### PR TITLE
Add more memory metric

### PR DESCRIPTION
**Pull request by OGKevin**
This closes https://jira.percona.com/browse/PMM-4348.

This should be merged after #25 I did it this way as I did not feel like dealing with merge conflicts afterwards.

### CODE REVIEW

File: proxysql_exporter.go [hunk 1]
```diff
@@ -37,10 +37,11 @@ var (
...
 	telemetryPathF = flag.String("web.telemetry-path", "/metrics", "Path under which to expose metrics.")
 
-	mysqlStatusF         = flag.Bool("collect.mysql_status", true, "Collect from stats_mysql_global (SHOW MYSQL STATUS).")
-	mysqlConnectionPoolF = flag.Bool("collect.mysql_connection_pool", true, "Collect from stats_mysql_connection_pool.")
-	mysqlConnectionListF = flag.Bool("collect.mysql_connection_list", true, "Collect connection list from stats_mysql_processlist.")
+	mysqlStatusF                 = flag.Bool("collect.mysql_status", true, "Collect from stats_mysql_global (SHOW MYSQL STATUS).")
+	mysqlConnectionPoolF         = flag.Bool("collect.mysql_connection_pool", true, "Collect from stats_mysql_connection_pool.")
+	mysqlConnectionListF         = flag.Bool("collect.mysql_connection_list", true, "Collect connection list from stats_mysql_processlist.")
 	mysqlDetailedConnectionListF = flag.Bool("collect.detailed.stats_mysql_processlist", false, "Collect detailed connection list from stats_mysql_processlist.")
+	memoryMetricsF               = flag.Bool("collect.stats_memory_metrics", true, "Collect memory metrics from stats_memory_metrics.")
```
Comment (OGKevin) [inline proxysql_exporter.go]: @AlekSi should this one also be made false?

File: exporter.go [hunk 2]
```diff
@@ -461,5 +468,92 @@ func scrapeDetailedMySQLConnectionList(db *sql.DB, ch chan<- prometheus.Metric)
...
+				prometheus.BuildFQName(namespace, "stats_memory", m.name),
+				m.help,
+				nil, nil,
+			),
+			m.valueType,
+			res.value,
+		)
+	}
+
+	return nil
```
Comment (OGKevin) [inline exporter.go]: return rows.err

File: proxysql_exporter.go (hunk 1 above)
Comment (AlekSi) [reply proxysql_exporter.go]: Yes, please.

**Comment by OGKevin** [comment]
@AlekSi I think this is is ready for 👀 when you have time 👍

File: exporter.go [hunk 3]
```diff
@@ -34,22 +34,23 @@ type Exporter struct {
...
 	scrapeErrorsTotal              *prometheus.CounterVec
 	lastScrapeError                prometheus.Gauge
 	lastScrapeDurationSeconds      prometheus.Gauge
 	proxysqlUp                     prometheus.Gauge
 }
 
-// NewExporter returns a new ProxySQL exporter for the provided DSN.
-// It scrapes stats_mysql_global, stats_mysql_connection_pool and stats_mysql_processlist if corresponding parameters are true.
-func NewExporter(dsn string, scrapeMySQLGlobal bool, scrapeMySQLConnectionPool bool, scrapeMySQLConnectionList bool, scrapeDetailedMySQLProcessList bool) *Exporter {
+// NewExporter It scrapes stats_mysql_global and stats_mysql_connection_pool if corresponding parameters are true.
```
Comment (AlekSi) [inline exporter.go]: I think there was a merge conflict on that line. The first line is lost, and the second one is outdated.

Comment (OGKevin) [reply exporter.go]: Ehh you're correct. Nice spot.

Comment (OGKevin) [reply exporter.go]: Fixed.

**Comment by OGKevin** [comment]
@AlekSi all yours buddy 👀

**Comment by AlekSi** [comment]
Thanks! 🚀

**Comment by OGKevin** [comment]
@AlekSi one small question, when can i expect a new release that includes the last 2 merged MR's ?

**Comment by AlekSi** [comment]
Do you need a binary release?

**Comment by OGKevin** [comment]
@AlekSi yes, would be nice, saves me the hassle of doing it my self in my forked repo. Then its just a matter of change the version number in the ansible play :).

**Comment by AlekSi** [comment]
I will cut a new release this week.

**Pull request merged**